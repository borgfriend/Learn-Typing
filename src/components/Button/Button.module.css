.button {
    background: red;
    color: white;
    padding: 0.5rem 0.75rem;
    font-weight: bold;
    cursor: pointer;
}

.button[color='default'] {
    background: blue;
}