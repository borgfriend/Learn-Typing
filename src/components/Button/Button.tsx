import * as React from 'react';
import styles from './Button.module.css';

type colors = 'default' | 'primary'

interface IButtonProps {
    onClick?: () => void;
    color?: colors;
}
/**
 * A Button Component
 * A button executes
 * @param props
 */
export const Button: React.SFC<IButtonProps> = ({ onClick, children, color = 'default' }) => {
    return (<button
        className={`${styles.button} ${styles[color]}`}
        onClick = { onClick }
        color={`${color}`}
        >
        { children }
    </button >);
}