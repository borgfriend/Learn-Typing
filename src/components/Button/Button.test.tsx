import { fireEvent, render } from "@testing-library/react";
import React from "react";
import { Button } from "./Button";
import {it, expect,vi } from "vitest"

it("renders a default button", () => {
  const { container } = render(<Button></Button>);
  expect(container.firstChild).toHaveStyle(`background-color: rgb(255, 0 , 0)`);
  expect(container).toMatchSnapshot();
});

it("renders a primary button", () => {
  const { container } = render(<Button color="primary">{"Label"}</Button>);
  expect(container.firstChild).toHaveStyle(`background-color: rgb(0, 0, 255)`);
  expect(container).toMatchSnapshot();
});

it("can interact with the button", () => {
  const mockFn = vi.fn();
  const { container } = render(
    <Button color="primary" onClick={mockFn}>
      {"Label"}
    </Button>
  );

  if (container.firstChild) {
    fireEvent.click(container.firstChild);
  }

  expect(mockFn).toBeCalled();
});
