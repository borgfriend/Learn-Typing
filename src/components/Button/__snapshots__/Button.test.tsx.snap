// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`renders a default button 1`] = `
<div>
  <button
    class="sc-braxZu wJQOW"
  />
</div>
`;

exports[`renders a primary button 1`] = `
<div>
  <button
    class="sc-braxZu sc-gJhJTp wJQOW dcSjso"
  >
    Label
  </button>
</div>
`;
